package/kedro_viz/server.py

```diff
@@ -501,11 +561,14 @@ def pipeline_data(pipeline_id):
 
     pipeline_node_ids = set()
     pipeline_nodes = []
+    modular_pipelines = set()
 
     for node in _DATA["nodes"]:
         if pipeline_id in node["pipelines"]:
             pipeline_node_ids.add(node["id"])
             pipeline_nodes.append(node)
+            if node["type"] == "task" and node.get("modular_pipelines"):
```

Comment:
Nodes that aren't tasks could be part of multiple modular pipelines (same dataset used in several modular pipelines), so the modular pipeline will only be part of the selected pipeline if it comes from a task node.

Reply:
Let's add this explanation as a comment in the code.

Reply:
Is that true though? The membership of a pipeline is entirely determined from the name of the dataset and cannot be shared between two hierarchies. Isn't it more correct to say that a node (task) can consume and produce datasets from/to different modular pipelines than its own?

Reply:
Ah yes your phrasing is much better. The thing I was mainly struggling with here is the list of modular pipelines that should be shown for the selected pipeline and I ended up only showing those which task nodes are part of and not the input/output of those nodes. I'll add a comment explaining this.